### PR TITLE
TDL 18803 Add SMS events streams

### PR DESCRIPTION
**Pull request by prijendev**
# Description of change
**TDL 18803 Add SMS events streams**

- Added below SMS events streams
  - Clicked SMS
  - Consented to Receive SMS
  - Failed to Deliver SMS
  - Failed to deliver Automated Response SMS
  - Received Automated Response SMS
  - Received SMS
  - Sent SMS
  - Unsubscribed from SMS
- Create a `shared` folder for common schemas
- Updated existing scheme to use shared schema.
- Updated integration test cases to include above streams

**TDL 18962 Update schema for campaign stream**
- Update json schema for `campaign` stream

# Manual QA steps
 - Run discover mode and verify schema and metadata for each stream in the catalog.
 - Run sync mode with the state to verify bookmark logic and fields in the record. 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_klaviyo/schemas/campaigns.json [hunk 1]
```diff
@@ -20,9 +20,7 @@
         "num_recipients": {
             "type": "integer"
         },
-        "lists": {
-            "$ref": "lists.json"
-        },
+        "lists": {},
```
Comment (prijendev) [inline tap_klaviyo/schemas/campaigns.json:23]: `lists` and `excluded_lists` are both fields that are not mentioned in the API document but API returns in the response. So, we are keeping the `empty` object datatype as the API documentation does not provide exact information.

**Review by savan-chovatiya — APPROVED** [review]

**Review by dbshah1212 — APPROVED** [review]

**Review by KrisPersonal — APPROVED** [review]

**Review by RushT007 — CHANGES_REQUESTED** [review]
Add assertion to check first sync and second sync respects the start date.

**Comment by prijendev** [comment]
> Add assertion to check first sync and second sync respects the start date.

We added one assertion to check the start date for the first sync and the last sync: [link](https://github.com/singer-io/tap-klaviyo/blob/master/tests/test_klaviyo_start_date.py#L106-#L121). Do we need to add anything else?

File: tests/test_klaviyo_start_date.py [hunk 2]
```diff
@@ -15,18 +15,33 @@ def name():
         return "tap_tester_klaviyo_start_date_test"
 
     def test_run(self):
+
+        # We are not able to generate test data so skipping two streams(mark_as_spam, dropped_email)
+        # Skipping given streams due to data available only on a single date -  failed_to_deliver_automated_response, failed_to_deliver
+        
+        expected_streams = self.expected_streams() - {"mark_as_spam", "dropped_email", "failed_to_deliver_automated_response",
```
Comment (kspeer825) [inline tests/test_klaviyo_start_date.py:22]: Since streams are being omitted here, can you remove the skip condition on line 96

**Review by kspeer825 — CHANGES_REQUESTED** [review]
Left one suggestion.

File: tests/test_klaviyo_start_date.py (hunk 2 above)
Comment (prijendev) [reply tests/test_klaviyo_start_date.py:22]: Removed the skip condition.

**Review by RushT007 — APPROVED** [review]

**Review by kspeer825 — APPROVED** [review]

**Review by luandy64 — APPROVED** [review]

**Pull request merged**